### ISSUE TITLE
[L] BCSC Setup

### ISSUE THREAD

**Issue by Dianadec** [bcgov/reserve-rec-api#33]
#### Description:

Next steps for BC Service Card integration. 

#### Acceptance Criteria:
As a public park user
When I go to log in to the system to book a camping reservation
Then I see a button that allows me to login with my BC Services Card.

#### Development Checklist:
- [x] pre-req: cognito is setup
- [ ] Set up BCSC in dev and test environments
- [ ] Login page shows option to click BC Services card like the following:
![Image](https://github.com/user-attachments/assets/6d83407e-9ae1-4131-b356-3b89c8f7370c)

no longer relevant (this was for DUP):
- [ ] ~New keycloak client created and configured correctly in each environment~

#### Dependencies

- Blocked by https://github.com/bcgov/reserve-rec-api/issues/36 

- Blocking

**Relevant documentation as reference**

**Definition of Ready**

- [ ] Acceptance criteria are included
- [ ] Wireframes are included (if applicable)
- [ ] Design / Solution is accepted by Product Owner (if applicable)
- [ ] Dependencies are identified (technical, business, regulatory/policy)
- [ ] Story has been estimated (under 13 pts)

**Definition of Done** 
- In progress:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] UI meets accessibility requirements
  - [ ] Unit tests are written
  - [ ] Work is traceable in GitHub
  - [ ] PR linked to ticket number
  - [ ] If needed/required - Dev adds flag/label to highlight any migration steps necessary prior to PROD deployment
- Code review:
  - [ ] Code is peer reviewed and has passed CI/CD tests
- QA:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Code is potentially shippable to the production environment
  - [ ] Functional features have been tested and passed by QA
  - [ ] UI components tested by designer
  - [ ] Code is deployed to PROD when moved to 'done' column (unless requested otherwise by PO)
- PO Review:
  - [ ] Acceptance criteria are tested (Functionality meets the acceptance criteria defined in the ticket)
  - [ ] Reviewed and approved by Product Owner

#### Notes:
- linked to ticket https://github.com/orgs/bcgov/projects/49/views/2?sliceBy%5Bvalue%5D=Reservations&pane=issue&itemId=84717072&issue=bcgov%7Creserve-rec-api%7C34
-

**Comment by LindsayMacfarlane** [comment]
Hi @marklise - please help refine this ticket :)

**Comment by Dianadec** [comment]
Refinement notes:
- Question: Are we using cognito for everything? or will we use keycloak as well?
  - Suggestion: yes, use cognito for all of reserve-rec 
  - Wes & team (BCSC) is in support of this as well.
  - Om also agrees - isn't a fan of the hybrid. Pick one or the other

**Comment by Dianadec** [comment]
Whoever pulls in this ticket, talk to Mark about the deets.

**Comment by Dianadec** [comment]
Sprint Planning notes:

Need to figure out which things we need to request eg: First name, last name, etc. - before we go to prod. Which things other than first name last name do we want, out of that person's profile. Keeping in mind - the more we ask for, the more we need to get into our PIA.
Confirm with business.
Can also change things no prob.

JM - Current system:
Fname, lname, address, email, phone number

**Comment by marklise** [comment]
Credentials received. Resuming.

**Comment by marklise** [comment]
Dev working - test will need to be checked/tweaked once we have a proper roll out.

**Comment by manuji** [comment]
This feature will be tested later when the environments are fully setup. Moving to PO review. No testing done.
